R/block.R

```diff
@@ -0,0 +1,106 @@
+#' Create a block of text to render as Markdown text in Quarto
+#'
+#' [qto_block()] based on [as_markdown()]
+#'
+#' @param ... dots to convert to character vector
+#' @inheritParams base::paste0
+#' @examples
+#' as_markdown(c("Hello world!"))
+#' @return character vector of length 1
+#' @seealso
+#' - [as_markdown()]
+#' - [knitr::asis_output()]
+#'
+#' @export
+qto_block <- function(..., sep = "", collapse = "", call = caller_env()) {
+  check_dots_unnamed(call = call)
+  structure(
+    paste0(..., sep = sep, collapse = collapse),
+    class = c("knit_asis", "quarto_block")
+  )
+}
+
+#' Create a Quarto div with optional classes, attributes, and other identifiers
+#'
+#' <https://quarto.org/docs/authoring/markdown-basics.html#divs-and-spans>
+#'
+#' @inheritParams qto_attributes
+#' @param .content If `.content` is supplied, any values passed to `...` are
+#'   ignored. If `.content` is `NULL`, it is set as all values passed to `...`.
+#' @param collapse Passed to [base::paste0()] with `.content.`
+#' @param drop_empty If `TRUE`, drop empty values from `.content` or `...`
+#' @param drop_na If `TRUE`, drop `NA` values from `.content` or `...`
+#' @param fence If numeric, `fence` must be a minimum of 3 and sets the number
+#'   of times the standard fence character ":" should be repeated. If character,
+#'   `fence` is used as is.
+#' @inheritParams qto_block
+#' @seealso
+#' - [div()]
+#' - [qto_callout_block()]
+#' @examples
+#'
+#' # div with an ID
+#' qto_div(
+#'   "This content can be styled with a border",
+#'   class = "border"
+#' )
+#'
+#' # Nested div
+#' qto_div(
+#'   qto_div("Here is a warning.", class = "warning"),
+#'   "More content.",
+#'   id = "special",
+#'   class = "sidebar",
+#'   fence = 5
+#' )
+#'
+#' @export
+qto_div <- function(...,
+                    id = NULL,
+                    class = NULL,
+                    css = NULL,
+                    .attributes = NULL,
+                    .content = NULL,
+                    collapse = "",
+                    fence = ":::",
```

Comment:
Oddly enough I recall that too. However, from the docs:

>  ... it can be helpful for _visual clarity_ to use fences of different lengths to distinguish nested divs from their parents

(emphasis mine)

I think your decision is a good compromise!